### DISCUSSION TITLE
Epochs for prior and unclip?

### DISCUSSION THREAD

**Discussion by user425846** [seruva19/kubin#96] [General]
Hi! Im currently training prior for a style with a large dataset (thousands of images). How many epochs would you recommend for both prior and unclip? Any recommendations how long i should be training these? Should i use the same epoch for both? Is it normal that the loss is not really decreasing while training prior?

Thanks for your help and for the awesome ui you guys built

**Comment by adrianpuiu** [comment]
I trained with 150 at the beginning and it came out ok, second try i used 120 and turned out ok to ...I did noticed some over cooking in the 150 one fut only when increasing the prior steps up to 100 . now it's up to you ... but from what i saw somewhere around @120 it's fine.

**Reply by user425846** [reply]
Oh wow 100+ epoch? Im training with 60.000 Midjourney V5 images right now, im curious if this will be good or just a waste of time, im just trying stuff out :D However one epoch takes me about 45 mins so 100 epochs is really a lot. Is that really necessary?

**Reply by adrianpuiu** [reply]
freaking hell, 60k , dude .... looool. and i was scraping the Mj site since few weeks ago , but you are way ahead of me :)))) .  In regards to epoch ,. its repetitions/image .. so yeah ... give me few minutes to re download the kandinsky 2.1 to make sure im not sayin bullshits

**Reply by adrianpuiu** [reply]
just checked the training notebooks.. yes , thats correct 120 epochs/image

**Comment by seruva19** [comment]
Personally I got not time no train something yet :) So I cannot tell for sure what are recommended settings. 100 seems like a nice starting point. But if you have so much images... Try around 30-40 maybe? And increase batch size to make it faster (if your GPU can cope with that).
It's mostly trial and error though.
I would like to add function to generate test image after each epoch, it would make it much easier to estimate training flow.

**Reply by user425846** [reply]
Running on A6000, 178 batch size is the max i was able to do. Do you know if i can somehow continue training after stopping it? Like if i only train 20 epoch prior and 20 epoch unclip and then go back to prior and train it further? Do you know how that would work?

**Reply by seruva19** [reply]
Yes, I assume that you can use undertrained checkpoints (that you get after each epoch) to resume training. Just point to its location instead of default path ("models/2_1/decoder_fp16.ckpt" in case of decoder).

**Comment by seruva19** [comment]
Concerning loss: try to increase learning rate, maybe the network convergence is too slow because of large dataset. There might be some other reasons though.. eventually we will work out best practices for training.

**Comment by adrianpuiu** [comment]
I invited Kodxana. maybe he can help you with the training on runpods

**Reply by kodxana** [reply]
I could though I do not have dataset access.

**Reply by user425846** [reply]
I used this dataset from huggingface https://huggingface.co/datasets/wanng/midjourney-v5-202304-clean, filtered by square aspect ration, resized the images. But i only downloaded 60k, there are more.

Also i used the dataset of the upscaled images only, as they will be better on average than random ones from midjourney

**Comment by adrianpuiu** [comment]
wait, how did you downloaded them ? cause  there on the link its only the
captioning files ...

theres also this one with 4Mil +

https://huggingface.co/datasets/tarungupta83/MidJourney_v5_Prompt_dataset/viewer/tarungupta83--MidJourney_v5_Prompt_dataset/train?row=6

**Reply by user425846** [reply]
The dataset i sent is based on the dataset you sent but it cleaned up the prompts, removing the parameters and such. I wrote a script to download the image, name it and create a txt file with the same name to save the caption. Its very rough and in C# .net tho so i dont know if its of much use for you

**Reply by adrianpuiu** [reply]
but i see the data is stored in a hadoop like format .... or apache parquet ...  i think i could do that with  using the  url-into-images module from python ...

**Reply by user425846** [reply]
I made a parser for parquet in .NET i could run the whole file through it and send it to you as csv. Would you be able to train the model with so many images? My training crashed through the night, dont know why...

**Reply by adrianpuiu** [reply]
neah, with that many i would also fail, the best option we have is [@kodxana](https://github.com/kodxana)
, he might have a better chance in runpod

**Reply by adrianpuiu** [reply]
@kodxana  please share your discord again . @user425846  could you try and contact him on discord  ? and share what you have so he can start garbing the files faster ?

**Reply by kodxana** [reply]
madiator2011

**Reply by user425846** [reply]
Sent you a request!

**Reply by seruva19** [reply]
@user425846 You said your training crashed, do you remember the error message?

**Comment by user425846** [comment]
Unfortunately not! I ran the training in a terminal within jupyter lab and
the next day all tabs were still open except the terminal, it had somehow
closed. So im not sure it even was an issue with the training script, modt
likely something with running it through jupyter labs

neutron_hare ***@***.***> schrieb am Sa. 10. Juni 2023 um
18:51:

**Comment by adrianpuiu** [comment]
@kodxana  @user425846  did you guys had any success with the training ? Will you share the final model ? :)
